### PR TITLE
Added sync when won challenge

### LINKED ISSUE #7716: Winner of challenge sees a notification for a previous challenge that they've won
Amazon_Warrior (4273f332-21e2-419a-8a35-e462ce341f5c) reports:
**I just won a Challenge-'Help the Poet'-but the alert that came up was for the very first Challenge I won 'A Weekend Fully Booked: 24-Hour Readathon'. The alert said I'd just won the Readathon Challenge, even though that happened ages ago.**

I've seen one other report of a similar problem and I was able to replicate the bug just now.

However I don't know if it happens always or just sometimes. It's probably the sort of thing that could happen always but few people report it.

_edit by Alys 2017-12-01:_

This bug is back. DCL ( 3bd49b39-283b-4d03-8117-7b87e9547e91 ) wrote in the Report a Bug guild: "when I closed a challenge (in the private guild I use to keep weekly to-do lists as challenges) and listed myself as the winner, the pop-up notification said I had won a different challenge! The challenge I closed was 'Weekly Work Tasks,' and the challenge it told me I won was 'Reflect on the Day...' which I have both in my personal guild and my party (and was the last challenge I won). The problem is just the pop-up, as the proper achievement is listed in my profile, and I got the gems I had associated with the challenge."

The code related to this will be different since the new website was released but the behaviour seems the same and the same style of fix might work, even though the tech is different, so I'm reopening this rather than creating a new issue. If other admins want to close an open a new one, feel free. :)

_edit by Alys 2020-04-14:_

It's back again. https://github.com/HabitRPG/habitica/issues/7716#issuecomment-613309165

### PR DESCRIPTION
**Pull request by TheHollidayInn**
Fixes #7716

### CODE REVIEW

**Comment by paglias** [comment]
I'm not sure this will fix the problem, because the notification is shown when a request is made and new notifications are returned, but when notifications are returned we also return the `userV` from the server and if the user is out of date a sync request is made automatically so this way we'll only be doing two

**Comment by TheHollidayInn** [comment]
I was about to ping you for this :P 

Yea, my only thinking is if you look at the PR from the old client, it did fix this problem by syncing. I still think there could be a delay in loading the user with the scenario you are describing, so the user could see the popup before the new version is loaded.

Although, maybe I should try check if the user:fetch is in progress?

**Comment by paglias** [comment]
Yeah i think the problem is the delay in syncing, but this won't change and actually the sync will start just slightly later since the modal has to be shown. But yeah I would try to reproduce this locally to see if the user is synced or the problem is another

File: website/client/components/achievements/wonChallenge.vue [hunk 1]
```diff
@@ -58,6 +58,12 @@ export default {
     };
   },
   methods: {
+    async syncUser () {
+      // @TODO: There are times when the user is not updated when this modal is shown
+      // It would be nice to not need to fetch to full user, maybe just the achievements?
+      // See https://github.com/HabitRPG/habitica/issues/7716
+      await this.$store.dispatch('user:fetch', {forceLoad: true});
```
Comment (negue) [inline website/client/components/achievements/wonChallenge.vue:65]: should something be executed after the dispatch? if not you can remove the `async`/`await` and it'll still work ^^

Comment (TheHollidayInn) [reply website/client/components/achievements/wonChallenge.vue:65]: Only if I return the promise and await this function.

**Pull request closed without merging**